app/src/main/kotlin/com/wire/android/ui/home/conversations/usecase/GetMessagesForConversationUseCase.kt

```diff
@@ -1,30 +1,52 @@
 package com.wire.android.ui.home.conversations.usecase
 
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import androidx.paging.flatMap
 import com.wire.android.mapper.MessageMapper
 import com.wire.android.ui.home.conversations.model.UIMessage
 import com.wire.android.util.dispatchers.DispatcherProvider
 import com.wire.kalium.logic.data.id.ConversationId
 import com.wire.kalium.logic.feature.conversation.ObserveUserListByIdUseCase
-import com.wire.kalium.logic.feature.message.GetRecentMessagesUseCase
+import com.wire.kalium.logic.feature.message.GetPaginatedFlowOfMessagesByConversationUseCase
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.flow.flowOn
 import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.mapLatest
 import javax.inject.Inject
 
-class GetMessagesForConversationUseCase
-@Inject constructor(
-    private val getMessages: GetRecentMessagesUseCase,
+class GetMessagesForConversationUseCase @Inject constructor(
+    private val getMessages: GetPaginatedFlowOfMessagesByConversationUseCase,
     private val observeMemberDetailsByIds: ObserveUserListByIdUseCase,
     private val messageMapper: MessageMapper,
     private val dispatchers: DispatcherProvider,
 ) {
 
     @OptIn(ExperimentalCoroutinesApi::class)
-    suspend operator fun invoke(conversationId: ConversationId): Flow<List<UIMessage>> =
-        getMessages(conversationId).flatMapLatest { messages ->
-            observeMemberDetailsByIds(messageMapper.memberIdList(messages))
-                        .map { userList -> messageMapper.toUIMessages(userList, messages) }
-                }.flowOn(dispatchers.io())
-        }
+    suspend operator fun invoke(conversationId: ConversationId): Flow<PagingData<UIMessage>> {
+        val pagingConfig = PagingConfig(
+            pageSize = PAGE_SIZE,
+            prefetchDistance = PREFETCH_DISTANCE,
+            initialLoadSize = INITIAL_LOAD_SIZE
+        )
```

Comment:
Couldn't this config be created somewhere so in case we call the usecase invokation more than once (🤷🏻‍♂️ ) it would not be spending its 1ms extra to create it?

Reply:
1. Creating a simple configuration object should not be a concern. 1ms is probably a thousand times more than it actually takes to do that 😅 
2. This also means storing this configuration somewhere in memory when it's not being used. So... pros and cons.